static/panes/stack-usage-view.ts

```diff
@@ -0,0 +1,177 @@
+        this.eventHub.emit('stackUsageViewOpened', this.compilerInfo.compilerId);
+    }
+
+    override getInitialHTML(): string {
+        return $('#stackusage').html();
+    }
+
+    override createEditor(editorRoot: HTMLElement): monaco.editor.IStandaloneCodeEditor {
+        return monaco.editor.create(
+            editorRoot,
+            extendConfig({
+                language: 'plaintext',
+                readOnly: true,
+                glyphMargin: true,
+            }),
+        );
+    }
+
+    override registerOpeningAnalyticsEvent() {
+        ga.proxy('send', {
+            hitType: 'event',
+            eventCategory: 'OpenViewPane',
+            eventAction: 'StackUsage',
+        });
+    }
+
+    override registerCallbacks() {
+        this.eventHub.emit('requestSettings');
+        this.eventHub.emit('findCompilers');
+
+        this.container.on('shown', this.resize, this);
+
+        const cursorSelectionThrottledFunction = _.throttle(this.onDidChangeCursorSelection.bind(this), 500);
+        this.editor.onDidChangeCursorSelection(e => {
+            cursorSelectionThrottledFunction(e);
+        });
+    }
+
+    override onCompileResult(id: number, compiler, result) {
+        if (this.compilerInfo.compilerId !== id || !this.isCompilerSupported) return;
+        this.editor.setValue(result.source);
+        if (result.hasStackUsageOutput) {
+            this.showStackUsageResults(result.stackUsageOutput);
+        }
+        // TODO: This is inelegant again. Previously took advantage of fourth argument for the compileResult event.
+        const lang = compiler.lang === 'c++' ? 'cpp' : compiler.lang;
+        const model = this.editor.getModel();
+        if (model != null && this.getCurrentEditorLanguage() !== lang) {
+            monaco.editor.setModelLanguage(model, lang);
+        }
+
+        if (!this.isAwaitingInitialResults) {
+            if (this.selection) {
+                this.editor.setSelection(this.selection);
+                this.editor.revealLinesInCenter(this.selection.startLineNumber, this.selection.endLineNumber);
+            }
+            this.isAwaitingInitialResults = true;
+        }
+    }
+
+    // Monaco language id of the current editor
+    getCurrentEditorLanguage() {
+        return this.editor.getModel()?.getLanguageId();
+    }
+
+    override getDefaultPaneName() {
+        return 'Stack Usage Viewer';
+    }
+
+    getDisplayableOpt(optResult: suCodeEntry) {
+        return {
+            // value: '**' + optResult.Qualifier + '** - ' + optResult.displayString,
+            value: optResult.displayString,
+            isTrusted: false,
+        };
+    }
+
+
+    showStackUsageResults(results: suCodeEntry[]) {
+        const opt: monaco.editor.IModelDeltaDecoration[] = [];
+
+        const groupedResults = _.groupBy(
+            /* eslint-disable-next-line @typescript-eslint/no-unnecessary-condition */ // TODO
+            results.filter(x => x.DebugLoc !== undefined),
```

Comment:
I think this filter (and eslint comment) can be eliminated, DebugLoc appears to always be set